src/app/preferences/general-preferences/general-preferences.html

```diff
@@ -2,12 +2,7 @@
   <form class="form-horizontal general-preferences-form">
     <div class="form-group">
       <label class="col-md-2 control-label" for="Enabling toggle">Hide vertical nav by default</label>
-      <div class="col-md-6">
-        <select id="vertical-nav" class="form-control" ng-model="defaultVerticalNavState"
-          ng-options="defaultVerticalNavState for defaultVerticalNavState in availableVerticalNavs.availableOptions"
-          ng-change="onToggleChange(defaultVerticalNavState)">
-        </select>
-      </div>
+      <input type="checkbox" ng-change="onToggleChange(defaultVerticalNavState)" ng-model="defaultVerticalNavState">
```

Comment:
This doesn't seem to implement the bootstrap switch precisely enough.
https://www.patternfly.org/v3/pattern-library/widgets/#bootstrap-switch
And you'd need to invoke this somewhere in the controller.
```
jQuery(".bootstrap-switch").bootstrapSwitch();
```

Also see how checkbox can be implemented with AngularJS PF:
https://patternfly.github.io/angular-patternfly/#!/api/patternfly.validation:pfValidation